core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -849,7 +849,7 @@ protected final void unparseFetchUsingLimit(SqlWriter writer, SqlNode offset,
    * {@code SELECT SUM(SUM(1)) }.
    */
   public boolean supportsNestedAggregations() {
-    return true;
+    return false;
```

Comment:
I think there are actually dbs that support this (e.g. Oracle), so I don't think we need to change the default - specific dialects that we know don't support it already return false (Postgres, Hsql, Mysql).

Reply:
same comment applies to the places in dialects where you removed the override

Reply:
I'd feel happier leaving it as `false` by default, and override it to be `true` for DBs where we know that they support it. We can then have a test to verify that `SELECT SUM(SUM(1))` actually works for those DBs.

Reply:
I tend to lean towards switching off features by default because it's more stable. For any new dialect you can just forget to redefine this. And there are already a number of such errors. Anyway, if database product supports nested aggregates probably it's wise enough to flatten several aggregations itself.

Reply:
To maintain compatibility with Calcite I don't think we can change this value. They most likely wouldn't accept the PR. With good reason if you think about it - if a 3rd party service has integrated with calcite and added their own dialects which rely on defaults in the base class, then these will now break.
I think it's much safer to keep the base functionality as is, and leave it up to the integrator of a new dialect to decide whether to override the defaults or not.

Reply:
We're pretty much guaranteed to forget though, when we implement a new dialect. That interface is overwhelming. I agree that upstream is not going to merge this, but I think it should be a non-goal to be 100% the same as upstream Calcite.

The cost for forgetting to disable this on a dialect basis (in terms of our time) is high. The cost of maintaining a 1 line diff against Calcite is low.

Reply:
Yep, not wise to introduce such flags with default = `true`. They already broke compatibility when this flag is introduced because they will not updated all dialects to correct value.

Reply:
>  I think it should be a non-goal to be 100% the same as upstream Calcite.

It might not be the highest priority, but I think it should absolutely be a goal! Why would we want to maintain a fork and deal with potential conflicts and applying patches every time there's a new version if we don't have to?

I think we might be able to get away with this change if we can convince ourselves (and hopefully the calcite maintainers) that having a dialect switch from a value of `true` to `false` won't actually break any queries. Now that I've thought about it some more that probably is the case. I'd guess for most dbs it would probably just be a performance issue, and knowing what we do about optimizers it might even be more efficient in some cases to switch from `true` to `false` where both are supported (I have no idea if there is a real world case). What do you think @pavelgubin ?

Reply:
When preparing patch for Calcite we can describe the case as best as we can and hopefully they accept because it's natural and more robust, most products doesn't have this functionality. Else, we will switch over the flag in order to be compliant.